docs/proposals/control-plane-config.md

```diff
@@ -17,3 +17,86 @@ option that we will support is log level.
 ## Non-Goals
 
 This proposal is *not* defining a way to dynamically configure the data plane.
+
+## Introduction
+
+The NKG control plane will evolve to have various user-configurable options. These could include, but are not
+limited to, log level, tracing, or metrics. For the best user experience, these options should be able to be
+changed at runtime, to avoid having to restart NKG. The first option that we will allow users to configure is the
+log level. The easiest and most intuitive way to implement a Kubernetes-native API is through a CRD.
+
+## API, Customer Driven Interfaces, and User Experience
+
+The API would be provided in a CRD. An authorized user would interact with this CRD using `kubectl` to `get`
+or `edit` the configuration.
+
+Proposed configuration CRD example:
+
+```yaml
+apiVersion: nginx.gateway.k8s.io/v1beta1
+kind: NGINXControlConfig
+metadata:
+    name: nkg-config
+    namespace: nginx-gateway
+spec:
+    logLevel: info
+    ...
+```
+
+- The CRD would be Namespace-scoped, living in the same Namespace as the controller that it applies to.
+- CRD is initialized and created when NKG is deployed
+- NKG references the name of this CRD via CLI arg, and only watches this CRD
+- If user deletes resource, NKG logs an error and creates an event. Last state is used until CRD is restored.
+
+For discussion with team:
+
+- API group name
+- kind name
+- default resource name
+
+## Use Cases
+
+The high level use case for dynamically changing settings in the NKG control plane is to allow users to alter
+behavior without the need for restarting NKG and experiencing downtime.
+
+For the specific log level use case, users may be experiencing problems with NKG that require more information to
+diagnose. These problems could include:
+
+- Not seeing or processing Kubernetes resources that it should be.
+- Configuring the data plane incorrectly based on the defined Kubernetes resources.
+- Crashes or errors without enough detail.
+
+Being able to dynamically change the log level can allow for a quick way to obtain more information about
+the state of the control plane without losing that state due to a required restart.
+
+## Testing
+
+Unit tests can be leveraged for verifying that NKG properly watches and acts on CRD changes. These tests would
+be similar in behavior as the current unit tests that verify Gateway API resource processing.
+
+## Security Considerations
```

Comment:
Can we add more details about validation? and extend this doc if necessary https://github.com/nginxinc/nginx-kubernetes-gateway/blob/main/docs/resource-validation.md 
Specifically, how we will perform validation? (CEL, Webhook, OpenAPI scheme? what about in the control plane process)